### PR TITLE
132: Sort search results by location

### LINKED ISSUE #132: Sort search results by location
If the location of the user is available, use it to sort the search results by location.
Otherwise use alphabetical sorting.

This only needs to be done in the frontend, as the backend endpoint is already developed in #127 .

**Additional context**
User story #118

### PR DESCRIPTION
**Pull request by Schedulaar**
Closes #132 

This should be tested on iOS

### CODE REVIEW

File: frontend/lib/search/results_loader.dart [hunk 1]
```diff
@@ -79,16 +79,24 @@ class ResultsLoaderState extends State<ResultsLoader> {
...
-                AcceptingStoresSearch$Query$AcceptingStore>(
-            itemBuilder: (context, item, index) =>
-                SearchResultItem(key: ValueKey(item.id), item: item)),
-        separatorBuilder: (context, index) =>
-            Divider(height: 0, color: Colors.grey),
-      );
+  Widget build(BuildContext context) {
+    if (widget.coordinates != null) {
+      print(widget.coordinates.lat);
+      print(widget.coordinates.lat);
```
Comment (maxammann) [inline frontend/lib/search/results_loader.dart:85]:
```suggestion
```

File: frontend/lib/search/search_page.dart [hunk 2]
```diff
@@ -51,17 +58,28 @@ class _SearchPageState extends State<SearchPage> {
...
   @override
   void initState() {
     _textEditingController = TextEditingController();
     _focusNode = FocusNode();
+    determinePosition()
+        .then((value) => setState(() => {
+              _coordinates =
+                  CoordinatesInput(lat: value.latitude, lng: value.longitude)
+            }))
+        .whenComplete(() => this.setState(() {
```
Comment (maxammann) [inline frontend/lib/search/search_page.dart:80]: Should we display an error when the lookup of the location "failed". If there is no permission then this is fine. But there could also be errors afaik.

File: frontend/pubspec.yaml [hunk 3]
```diff
@@ -23,6 +23,7 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
+  intl: ^0.16.1
```
Comment (maxammann) [inline frontend/pubspec.yaml:26]: I did not spot the usage of this. Was this added by purpose?

**Review by maxammann — APPROVED** [review]

File: frontend/pubspec.yaml (hunk 3 above)
Comment (Schedulaar) [reply frontend/pubspec.yaml:26]: Yep, it's used for NumberFormat (formatting the distance)

File: frontend/lib/search/search_page.dart (hunk 2 above)
Comment (Schedulaar) [reply frontend/lib/search/search_page.dart:80]: Hmmm.. yeah, the current fallback is using alphabetic sorting of the results :thinking: Could definitely be polished

File: backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/database/repos/AcceptingStoresRepository.kt [hunk 4]
```diff
@@ -8,7 +8,7 @@ import xyz.elitese.ehrenamtskarte.webservice.schema.types.Coordinates
 object AcceptingStoresRepository {
 
     fun findByIds(ids: List<Int>) =
-        AcceptingStoreEntity.find { AcceptingStores.id inList ids }.associateWithKeys({ it.id.value }, ids)
+        AcceptingStoreEntity.find { AcceptingStores.id inList ids }.sortByKeys({ it.id.value }, ids)
 
 
     // TODO would be great to support combinations like "Tür an Tür Augsburg"
```
Comment (aiquita) [inline backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/database/repos/AcceptingStoresRepository.kt:14]: Jap, would be nice 👍 
On the other hand the code is not a issue board.

File: backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/database/repos/AddressRepository.kt [hunk 5]
```diff
@@ -2,9 +2,9 @@ package xyz.elitese.ehrenamtskarte.database.repos
 
 import xyz.elitese.ehrenamtskarte.database.AddressEntity
 import xyz.elitese.ehrenamtskarte.database.Addresses
-import xyz.elitese.ehrenamtskarte.database.associateWithKeys
+import xyz.elitese.ehrenamtskarte.database.sortByKeys
 
 object AddressRepository {
     fun findByIds(ids: List<Int>) =
-        AddressEntity.find { Addresses.id inList ids }.associateWithKeys({ it.id.value }, ids)
+        AddressEntity.find { Addresses.id inList ids }.sortByKeys({ it.id.value }, ids)
```
Comment (aiquita) [inline backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/database/repos/AddressRepository.kt:9]: How does this lead to sorting by location or is this done in frontend?

Comment (Schedulaar) [reply backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/database/repos/AddressRepository.kt:9]: It doesn't. We only renamed `associateWithKeys` to `sortByKeys`

File: backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/database/repos/AcceptingStoresRepository.kt (hunk 4 above)
Comment (Schedulaar) [reply backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/database/repos/AcceptingStoresRepository.kt:14]: I didn't add the todo

**Pull request merged**